### DISCUSSION TITLE
English as main language in repo

### DISCUSSION THREAD

**Discussion by loxy** [taiga-family/taiga-ui#5741] [General]
Hi,

I see issues and discussion from time to time which are written in Russian. This is not good for people how do not understand that language. It is hard then to participate in those conversations.

What do you think?

Best regards,
Kersten

**Comment by waterplea** [comment]
What would you suggest? Some people just come here from Russian speaking communities and ask their questions in Russian, knowing we can understand them. I would also prefer they used English so everybody can participate, but I don't see an option to enforce this rule.

**Reply by loxy** [reply]
I understand that. I would also love to use German. If you would understand German, would I'm allowed to use it?

The whole code base is in English and your documentation as well. How do these people read it? How do these people read all the other issues and discussions? Do they not search before filling up a new item?

Nowadays there are endless possibilities to translate things. Software is now really good at that. And I can't imagine how one can work in this area without any knowledge of English. But anyway...

So, what about make it somehow clear to use or prefer English? Let's put it to the templates or mention it on the discussion page. You as the maintainers should make clear, that this is an international project with people contributing from all over the world. 

And if someone adds again an Russian issue, it would be great if you can kindly point out this fact in one of the answers.

This would be my suggestion.

**Reply by omqmo** [reply]
Очень провакационное предложение. Багов на английском языке более чем достаточно чтобы реагировать на русскую речь. Хотя меня тоже несколько лет назад этот факт забавлял. Но я привык. Даже удобно. Если бы я знал немецкий, то вообще бы не был против помочь вам, так как это лишний повод подтянуть свои знания. 

И кстате, это первый и кажется единственный репозиторий в котором я русскую речь лицезрею. Китайской тоже много.

**Reply by loxy** [reply]
> That's a very provocative suggestion. There are more than enough bugs in English to respond to Russian speech. Although I was also amused by this fact a few years ago. But I got used to it. It's even comfortable. If I knew German, I wouldn't mind helping you at all, as it's an extra reason to tighten up my knowledge.

> And by the way, this is the first and I think the only repository where I see Russian speech. There's a lot of Chinese too.

That is what https://www.deepl.com/translator made out of it...

So, you understand. Or at least, are able to understand. QED.

But I don't understand your arguments. The only one is: There are so little repos out there which use Russian, we should stick to it further to keep that language alive?

And if someone has problems and searches for hints he should just use search terms in both languages. Makes sense...

And by the way, there are more Chinese than European and Russian together

**Reply by waterplea** [reply]
I wouldn't want to discourage our users. Our library is popular in Russian speaking companies because we do a lot of tech writing and meetups in Russian and have a Russian speaking support chat. Some of those issues are just filled by product devs who use Russian in internal bug trackers and interfaces and they didn't stop to think this might be a problem for somebody. Most people do use English though, as they see everything here is written in English. I'm sorry it causes you discomfort but I don't think this is something we should enforce.

@omqmo I don't think it makes sense to purposefully use Russian in a discussion like that either. Sounds rude. I don't think there's a clear way to make "international" and "Russian speaking" distinction on GitHub, but on our Telegram we have separate topics for languages:
https://t.me/taiga_ui

Most of our international users ask questions on Angular Discord though, so I'll drop a link to it here as well, it is strictly English:
https://discord.gg/angular

**Reply by loxy** [reply]
Ok. Understood.

But you want to say that all of the last discussions (first page) in Russian:

- https://github.com/orgs/taiga-family/discussions/5597
- https://github.com/orgs/taiga-family/discussions/5510
- https://github.com/orgs/taiga-family/discussions/5444
- https://github.com/orgs/taiga-family/discussions/5092
- https://github.com/orgs/taiga-family/discussions/4636

are not able to use English? 20%?

For me it looks like some of them are not willing (like @omqmo) to use it.

I don't say you should enforce it or something like that. But a kindly reminder on some of the pages to inform people and make them aware are also not a big deal, right? To get more in the direction of 5%...

I mean, at the end I don't care. I just don't answer issues which I don't understand and also don't take them into account when searching for stuff. You as the maintainers need to answer most of the questions twice in such cases.
